tldw_Server_API/tests/Chat/unit/test_model_router_accounting.py

```
@@ -0,0 +1,59 @@
+    assert payload["router_selected_provider"] == "openai"
+    assert payload["router_selected_model"] == "gpt-4.1"
+    assert payload["execution_provider"] == "openai"
+    assert payload["execution_model"] == "gpt-4.1-mini"
+    assert payload["fallback_used"] is True
+
+
+@pytest.mark.asyncio
+async def test_log_model_router_usage_records_router_operation():
+    calls: list[dict[str, object]] = []
+
+    async def fake_usage_logger(**kwargs):
```

Ruff (0.15.6)

[warning] Missing return type annotation for private function `fake_usage_logger`

Add return type annotation: `None`

(ANN202)